### DISCUSSION TITLE
Why are some files missing from mounted volumes?

### DISCUSSION THREAD

**Discussion by mcepl** [containers/podman#22809] [Q&A] [answered]
Trying to run [dovecot IMAP server](https://www.dovecot.org/) as a rootless podman container via Quadlet with this `.container` file (using podman 5.0.3 on openSUSE/MicroOS):

```
mitmanek:~ $ cat ~/.config/containers/systemd/dovecot.container
[Unit]
Description=Podman dovecot IMAP server container
Documentation= https://github.com/thkukuk/containers-mailserver/blob/master/dovecot/README.md
Wants=network-online.target
After=network-online.target

[Container]
Image=registry.opensuse.org/opensuse/dovecot
ContainerName=dovecot
HostName=my-dovecot
#Network=host
AutoUpdate=registry
Environment=DOVECOT_TLS=0 \
	USE_LDAP=0 TZ='Europe/Prague' \
	ENABLE_LMTP=1
PublishPort=1143:143
# Variables like %h don't seem to work here
Volume=/home/matej/.config/dovecot:/etc/dovecot:z
Volume=/etc/ssl:/etc/certs
Volume=/home:/var/mailstore
# Volume=/var/mail:/var/mail:z
# UserNS=keep-id:uid=1000,gid=100

[Service]
Restart=on-failure

[Install]
WantedBy=default.target
```

I have this in my local configuration folder:

```
mitmanek:~ $ ls /home/matej/.config/dovecot
conf.d                      dovecot-ldap.conf.ext    local.conf
dovecot.conf                dovecot-oauth2.conf.ext  sieve-ldap.conf
dovecot-dict-auth.conf.ext  dovecot-sql.conf.ext     ssl
dovecot-dict-sql.conf.ext   imap.passwd
mitmanek:~ $ 
```

However, when I look into the running container both `local.conf` and `imap.passwd` are missing:

```
mitmanek:~ $ podman run -it dovecot /usr/bin/ls /etc/dovecot/
conf.d                      dovecot-ldap.conf.ext       dovecot.conf
dovecot-dict-auth.conf.ext  dovecot-oauth2.conf.ext     sieve-ldap.conf
dovecot-dict-sql.conf.ext   dovecot-sql.conf.ext
mitmanek:~ $ 
```

What I am doing wrong?

**Comment by Luap99** [comment, marked as answer]
If you use podman run you create a new container and you do not mount the volume there at all so it just shows you the image content. If you want to execute commands in an existing running container you have to use `podman exec`

**Reply by mcepl** [reply]
You are right, I was too stubborn to notice it.

**Comment by mcepl** [comment]
On Mon May 27, 2024 at 11:22 AM CEST, Paul Holzinger wrote:
> If you use podman run you create a new container and you do not mount the volume there at all so it just shows you the image content. If you want to execute commands in an existing running container you have to use `podman exec`

I don’t understand: that cannot be correct. When I start the container with `systemctl --user start` (or `systemctl --user daemon-reexec`), `podman run` with that container still doesn’t show the missing files.

**Reply by Luap99** [reply]
Your example shows `podman run` without any volumes.
.container files use the volumes you configured but if you want to check that you must use this container to check and not create a new one with podman run
